sbysrc/sby_core.py

```diff
@@ -287,10 +292,61 @@ def parse_config(self, f):
 
                 if section == "engines":
                     mode = "engines"
-                    if len(self.engines) != 0:
-                        self.error(f"sby file syntax error: '[engines]' section already defined")
+
                     if args is not None:
-                        self.error(f"sby file syntax error: '[engines]' section does not accept any arguments. got {args}")
+                        section_args = args.split()
+
+                        if len(section_args) > 1:
+                            self.error(f"sby file syntax error: '[engine]' sections expects at most 1 argument, got '{len(section_args)}'")
+
+                        if section_args[0] not in ("bmc", "prove", "cover", "live"):
+                            self.error(f"sby file syntax error: Expected one of 'bmc, prove, cover, live' as '[engine]` argument, not '{section_args[0]}'")
+
+                        if section_args[0] in self.engines:
+                            self.error(f"Already defined engine block for mode '{section_args[0]}'")
+                        else:
+                            self.engines[section_args[0]] = list()
+                            engine_mode = section_args[0]
+                    continue
+
+                if section == "setup":
+                    self.error(f"sby file syntax error: the '[setup]' section is not yet supported")
+
+                    mode = "setup"
+                    if len(self.setup) != 0:
+                        self.error(f"sby file syntax error: '[setup]' section already defined")
+
+                    if args is not None:
+                        self.error(f"sby file syntax error: '[setup]' section does not accept any arguments. got {args}")
+
+                    continue
+
+                # [stage <NAME> (PARENTS,...)]
+                if section == "stage":
+                    self.error(f"sby file syntax error: the '[stage]' section is not yet supported")
+
+                    mode = "stage"
+
+                    if args is None:
+                        self.error(f"sby file syntax error: '[stage]' section expects arguments, got none")
+
+                    section_args = args.strip().split(maxsplit = 1)
+
+
+                    if len(section_args) == 1:
+                        parents = None
+                    else:
+                        parents = list(map(lambda a: a.trim(), section_args[1].split(',')))
```

Comment:
I don't really mind this, but `list(map(lambda a: a.strip(), section_args[1].split(',')))` is the same as `[a.strip() for a in section_args[1].split(',')]` which I think is a bit simpler. Either is fine, IMO, just noticed this because I used to instinctively write `map(lambda a: ..., ...)` a lot (and still do from time to time) while I actually prefer list-comprehension so this is something that sticks out to me now.

Reply:
No need to change it then, which one to chose doesn't affect any code anywhere else, so I don't see the need to make consistent choices for this